lib/temporal/client.rb

```diff
@@ -0,0 +1,382 @@
+require 'json'
+require 'securerandom'
+require 'socket'
+require 'temporal/errors'
+require 'temporal/converter'
+require 'temporal/interceptor/client'
+require 'temporal/interceptor/chain'
+require 'temporal/workflow/handle'
+require 'temporal/workflow/execution_info'
+require 'temporal/workflow/id_reuse_policy'
+require 'temporal/workflow/query_reject_condition'
+
+module Temporal
+  class Client
+    attr_reader :namespace, :identity
+
+    # TODO: More argument to follow for converters, codecs, etc
+    def initialize(connection, namespace, interceptors: [])
+      @connection = connection
+      @namespace = namespace
+      @identity = "#{Process.pid}@#{Socket.gethostname}"
```

Comment:
It's helpful when debugging histories for the identity to include the fact that this is the ruby sdk and its version by default

Reply:
Added, please let me know if there's a format to follow